staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/client_test.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"testing"
+
+	"k8s.io/api/admissionregistration/v1beta1"
+)
+
+func TestHookClient(t *testing.T) {
+	scenarios := []struct {
+		clientConfig v1beta1.WebhookClientConfig
```

Comment:
Could you also add some test case for `clientConfig` using `Service` instead of `URL`?